keps/sig-node/3063-dynamic-resource-allocation/README.md

```diff
@@ -748,6 +747,43 @@ Some of the race conditions that need to be handled are:
   put the pod back into the queue, waiting for the ResourceClaim to become
   usable again.
 
+- Two pods get created which both reference the same unallocated claim with
+  delayed allocation. A single scheduler could detect this special situation
+  and then trigger allocation only for one of the two pods. But it is simpler
+  to proceed with pod scheduling for both of them independently, which implies
+  trying to select a node and allocate for it in parallel. Depending on timing,
+  the resource driver will see one of the requests for allocation first and
+  execute it. The other pod then either can share the same resource (if
+  supported) or must wait until the first one is done with it and reallocate
+  it.
+
+- Scheduling a pod and allocating resources for it has been attempted, but one
+  claim needs to be reallocated to fit the overall resource requirements. A second
+  pod gets created which references the same claim that is in the process of
+  being deallocated. The claim seems to be available (allocated, no other users),
+  so the scheduler reserves it for the second pod. In the meantime, the driver
+  finishes deallocation.
+
+  <<[UNRESOLVED pohly]>>
+  The scheduler knows about all pods (assuming a single scheduler per cluster). Should
+  it detect the pending deallocation request? That only works under the assumption
+  that only pods are users for which a claim might get reserved.
+
+  The driver could clear the `Allocated` field before actually finishing the
+  deallocation. Then the claim will not be used for the second pod, but the
+  scheduler also cannot tell anymore whether deallocation has completed
+  successfully and may start the next scheduling attempt too soon. We could
+  consider the `DeallocationRequested` field, but if pod and PodScheduling
+  get removed, we still have the problem should the claim get referenced again.
+
+  Instead of using the `DeallocationRequested` field, the scheduler could
+  delete the ResourceClaim. It will then get deallocated and removed, at which point
+  the resource controller will recreate it for inline resources. This is a slower
+  process, but it would work. But this does not work for user-created ResourceClaims.
+  Do we prevent delayed allocation for those? How?
+  See also https://github.com/kubernetes/enhancements/pull/3064#discussion_r902133513
+  <<[/UNRESOLVED]>>
```

Comment:
My preferred approach for solving this race is to keep `DeallocationRequested` in the `ResourceClaimStatus`. It is used during scheduling, but could also serve other purposes.